apps/api/src/app/events/services/mail-service/handlers/msgraph.handler.ts

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { ChannelTypeEnum } from '@novu/shared';
+import { MSGraphAPIProvider } from '@novu/msgraph';
+import { BaseHandler } from './base.handler';
+import { ICredentials } from '@novu/dal';
+
+export class MSGraphHandler extends BaseHandler {
+  constructor() {
+    super('msgraph', ChannelTypeEnum.EMAIL);
+  }
+  buildProvider(credentials: ICredentials, from?: string) {
+    // null-assertion is disabled because we want to expose the error from the MS Graph API
```

Comment:
In release 0.9.0 that's almost done we add a feature called Advanced Debugging that for every notification will show the steps taken by every notification, which parts of the steps failed and the reason why, so this wouldn't be needed in my opinion. Unless I am missing any nuance.